### DISCUSSION TITLE
Trying to Create Helper for Manual Plugins

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/PluginManager#36] [Q&A] [answered]
So if the `remote_install` is set to false, the plugin doesn't display at all.

https://kanboard.discourse.group/t/new-plugin-pluginmanager/2492/35?u=aljawaid

### I would create a helper, `foreach` to show all plugins set as `remote_install = false` 

Calling:
```php
<p class="">
<?php foreach ($this->helper->pluginManagerHelper->getAllPlugins() as $manualPlugin): ?>
    <?php if ($manualPlugin['remote_install'] == false): ?>
        <?= $this->text->e($manualPlugin['title']) ?>,
    <?php endif ?>
<?php endforeach ?>
</p>
```

Helper:
```php
public function getAllPlugins()
    {
        $url = PLUGIN_API_URL;
        $getJson($url, array $headers = [], $raiseForErrors = false)

        $response = $this->doRequest('GET', $url, '', array_merge(['Accept: application/json'], $headers), $raiseForErrors);
        return json_decode($response, true) ?: [];


        $all_plugins = $getJson($url);
        $manual_plugins = Kanboard\Core\Plugin\Directory::getInstance($this->container)->isInstallable($all_plugins);
    }
```

Can anybody help to fix this? @creecros @alfredbuehler 

I can't seem to find a function for the `directory.php` or even the `show.php` which finds ALL plugins... all the functions seem to point to `is installable` and `is configured` but I just want ALL the plugins and then show the ones which have remote install set to false.

**Comment by aljawaid** [comment, marked as answer]
I think I might have sorted it... standby caller

**Reply by aljawaid** [reply]
```php
public function getAllPlugins($url = PLUGIN_API_URL)
    {
        $manual_plugins = $this->httpClient->getJson($url);
        $manual_plugins = array_filter($manual_plugins, array($this, 'isNotInstallable'));
        return $manual_plugins;

    }

    public function isNotInstallable(array $plugin)
    {
        return $plugin['remote_install'] == false;
    }
```

very rough draft:
```php
<p class="">
<?php foreach ($this->helper->pluginManagerHelper->getAllPlugins() as $manualPlugin): ?>

        <?= $this->text->e($manualPlugin['title']) ?>,

<?php endforeach ?>
</p>
```

**Comment by alfredbuehler** [comment]
> I just want ALL the plugins and then show the ones which have remote install set to false.

What will be the purpose of all this?

**Reply by aljawaid** [reply]
So basically this will create a separate section 'manual updates' which will just LIST the plugins if they are set for remote_install=false.

The listed plugins will have links to their repositories and releases, but not be allowed to be installed directly.

So, like the skeleton plugin... which is not a functioning plugin... users normally would not find it.  It would be in the manual section.  The same for a few other plugins like th ones posted by https://kanboard.discourse.group/t/various-plugins-tagi-plugins/2683/8

**Reply by aljawaid** [reply]
![image](https://user-images.githubusercontent.com/10233708/225151109-aa2c1ef0-fafb-4535-94d7-6b9080153ab4.png)

A very very rough draft

**Reply by aljawaid** [reply]
I will move it to a new page

**Reply by aljawaid** [reply]
a bit of a description I put in the readme:
```text
- NEW: Manual Plugins Page
  - Show a list of plugins which can be manually downloaded and installed
  - Manual Plugins are part or completely functional but not suitable for general Kanboard installation
  - These plugins only show in PluginManager
```

**Reply by aljawaid** [reply]
![image](https://user-images.githubusercontent.com/10233708/225433942-2410982f-0654-423a-b0e5-2db8e2a14c12.png)
